src/Init/Data/Nat/Basic.lean

```diff
@@ -750,12 +824,21 @@ theorem succ_pred {a : Nat} (h : a ≠ 0) : a.pred.succ = a := by
   | zero => contradiction
   | succ => rfl
 
+theorem add_one_add_one {a : Nat} (h : a ≠ 0) : a - 1 + 1 = a := by
+  induction a with
+  | zero => contradiction
+  | succ => rfl
+
 theorem succ_pred_eq_of_pos : ∀ {n}, 0 < n → succ (pred n) = n
   | _+1, _ => rfl
 
 theorem sub_one_add_one_eq_of_pos : ∀ {n}, 0 < n → (n - 1) + 1 = n
   | _+1, _ => rfl
 
+theorem eq_zero_or_eq_add_one_sub_one : ∀ {n}, n = 0 ∨ n = (n - 1) + 1
```

Comment:
Name.

Reply:
I'm not sure what the name should be, this is completely analogous to `Nat.eq_zero_or_eq_succ_pred`, which follows the usual mathematical application `(· + 1) ((· - 1) n)`, which is the opposite of how it's written `n - 1 + 1`.

Reply:
[Nat.sub_one_add_one_eq_of_pos](https://leanprover-community.github.io/mathlib4_docs/Init/Data/Nat/Basic.html#Nat.sub_one_add_one_eq_of_pos) follows the same logic

Reply:
I see that [Nat.sub_one_add_one_eq_of_pos](https://leanprover-community.github.io/mathlib4_docs/Init/Data/Nat/Basic.html#Nat.sub_one_add_one_eq_of_pos) follows the other logic, so I will use it as well.

Reply:
Yes, when naming we read arithmetic from left to right, rather than making up functions to apply. :-)